webhook/server/server.go

```diff
@@ -90,12 +94,45 @@ func (s *WebhookServer) listenAndServe(client *client.Client, handler http.Handl
 		// Sleep here to make sure server is listening and all caches are primed
 		time.Sleep(15 * time.Second)
 
-		logrus.Debugf("Building validation rules...")
+		port := int32(types.DefaultWebhookServerPort)
+
+		logrus.Infof("Building conversion rules...")
+		for _, name := range conversionResources {
+			crd, err := client.CRD.CustomResourceDefinition().Get(name, metav1.GetOptions{})
+			if err != nil {
+				return secret, err
+			}
+
+			existingCRD := crd.DeepCopy()
+			crd.Spec.Conversion = &apiextv1.CustomResourceConversion{
+				Strategy: apiextv1.WebhookConverter,
+				Webhook: &apiextv1.WebhookConversion{
+					ClientConfig: &apiextv1.WebhookClientConfig{
+						Service: &apiextv1.ServiceReference{
+							Namespace: s.namespace,
+							Name:      webhookServiceName,
+							Path:      &conversionPath,
+							Port:      &port,
+						},
+						CABundle: secret.Data[corev1.TLSCertKey],
+					},
+					ConversionReviewVersions: []string{"v1beta2", "v1beta1"},
+				},
+			}
```

Comment:
It seems that we can directly add this field in the CRD YAML file rather than updating it after the installation:
```
conversion:
    strategy: Webhook
    webhook:
      clientConfig:
        service:
          name: longhorn-webhook
          namespace: longhorn-system
          path: "/v1/webhook/conversion"
          port: 9443
      conversionReviewVersions:
        - v1beta1
        - v1beta2
```

I am not sure which one is better. What do you think?

Reply:
The `caBundle` is the CA certificate which is a byte slice.

I follow the pattern that how we register the ValidatingWebhookConfiguration and MutatingWebhookConfiguration.

Reply:
OK. CA certificate setting may be a problem for the YAML file. Let's retain the current implementation.